src/OpenTelemetry.Api/Context/Propagation/BaggagePropagator.cs

```diff
@@ -143,20 +143,31 @@ internal static bool TryExtractBaggage(string[] baggageCollection, out Dictionar
                         break;
                     }
 
-                    if (NameValueHeaderValue.TryParse(pair, out NameValueHeaderValue baggageItem))
+                    if (pair.IndexOf('=') < 0)
                     {
-                        if (string.IsNullOrEmpty(baggageItem.Name) || string.IsNullOrEmpty(baggageItem.Value))
-                        {
-                            continue;
-                        }
+                        continue;
+                    }
+
+                    var parts = pair.Split(new[] { '=' }, 2);
+                    if (parts.Length != 2)
+                    {
+                        continue;
+                    }
 
-                        if (baggageDictionary == null)
-                        {
-                            baggageDictionary = new Dictionary<string, string>();
-                        }
+                    var key = WebUtility.UrlDecode(parts[0]);
```

Comment:
Ideally (for best perf) we probably want to use [UrlDecode with offset](https://docs.microsoft.com/en-us/dotnet/api/system.web.httputility.urldecode?view=net-5.0#System_Web_HttpUtility_UrlDecode_System_Byte___System_Int32_System_Int32_System_Text_Encoding_). Not a blocker and can be a separate PR.

Reply:
Could you please give me a doc to read why it's better for performance?

Reply:
Also, if the UrlDecode is better for perf, is it means also that UrlEncode will be better during the injection process?

Reply:
I don't know a doc, but here goes the idea:

Instead of trying to split the string into smaller strings (which makes GC busy), we can potentially use an integer to store the index of `=` during the loop, and use the in-place version of UrlDecode, so that we avoid creating many intermediate strings.

This is the current code:

```csharp
foreach (var pair in item.split(',')) // this creates an enumerator, an array and several strings
{
  var parts = pair.split('=', ...); // this creates one array and two strings, for each loop
  decode(parts[0]); // this creates one result string
  decode(parts[1]); // this creates one result string
}
```

What we can potentially do:

```csharp
while (i < item.Length) // we only need few integers on the stack, no heap allocation needed for any intermediate value
{
  check if we got `=`, if yes, decode the key and remember the position
  check if we got the splitter `,`, if yes, decode the value with decode(item, startPosition, stopPosition)
}
```

Reply:
Good point. I've got the idea. As you said before it can be a separate PR.